packages/aws-sdk-client-mock-jest/src/jestMatchers.ts

```diff
@@ -217,317 +103,207 @@ interface AwsSdkJestMockAliasMatchers<R> extends Record<string, Function> {
-        input: Record<string, unknown>,
-        ...other: unknown[]
-    ) {
-        ensureNoOtherArgs(other);
-        assert(
-            call && typeof call === 'number' && call > 0,
-            'Call number must be a number greater than 0',
-        );
-
-        return processMatch<{ received: AnyCommand | undefined }>({
-            ctx: this,
-            mockClient,
-            command,
-            check: ({calls}) => {
-                if (calls.length < call) {
-                    return {pass: false, data: {received: undefined}};
-                }
-
-                const received = calls[call - 1].args[0];
-
-                let pass = false;
-                if (received instanceof command) {
-                    pass = this.equals(received.input, expect.objectContaining(input));
-                }
-
-                return {
-                    pass,
-                    data: {received},
-                };
-            },
-            message: ({cmd, client, data, notPrefix}) => [
-                `Expected ${client} to ${notPrefix}receive ${call}. ${cmd} with ${this.utils.printExpected(input)}`,
-                ...(data.received
-                    ? [
-                        `${client} received ${this.utils.printReceived(data.received.constructor.name)} with input:`,
-                        this.utils.printDiffOrStringify(
-                            input,
-                            data.received.input,
-                            'Expected',
-                            'Received',
-                            false,
-                        ),
-                    ]
-                    : []),
-            ],
-        });
-    },
-    /**
-     * implementation of {@link AwsSdkJestMockMatchers.toHaveReceivedNthSpecificCommandWith} matcher
-     */
-    toHaveReceivedNthSpecificCommandWith(
-        this: MatcherContext,
-        mockClient: unknown,
-        call: number,
-        command: new () => AnyCommand,
-        input: Record<string, unknown>,
-        ...other: unknown[]
-    ) {
-        ensureNoOtherArgs(other);
-        assert(
-            call && typeof call === 'number' && call > 0,
-            'Call number must be a number greater than 0',
-        );
-
-        return processMatch<{ received: AnyCommand | undefined }>({
-            ctx: this,
-            mockClient,
-            command,
-            check: ({commandCalls}) => {
-                if (commandCalls.length < call) {
-                    return {pass: false, data: {received: undefined}};
+export function createBaseMatchers<T extends CommonMatcherUtils = CommonMatcherUtils>(
+    errorMsg: MessageImpl<T>,
+    objectContaining: AsymmetricMatchers['objectContaining']
+): { [P in keyof AwsSdkMockBaseMatchers<unknown>]: MatcherFunction<T> } {
+    errorMsg.toHaveReceivedCommand;
+    return {
+        /**
+         * implementation of {@link AwsSdkJestMockMatchers.toHaveReceivedCommand} matcher
```

Comment:
`AwsSdkJestMockMatchers` no longer exists, should be `AwsSdkMockBaseMatchers` now (here and below). Or we can get rid of those comments, they do not have much value.